### PR TITLE
fix(material/schematics): fix card tmpl migration

### PR DESCRIPTION
**Pull request by wagnermaciel**
fixes https://github.com/angular/components/issues/26038

### CODE REVIEW

**Review by devversion — APPROVED** [review]
LGTM

**Comment by wagnermaciel** [comment]
@devversion This actually doesn't work 100% of the time. Specifically, `*ngFor` does not appear in the `inputs`, `outputs`, or `attributes` of a `TmplAstElement`. I'm going to take another stab at this on Monday

**Comment by devversion** [comment]
Oh yeah, good point. That will be a little more tricky since the template node is wrapping the element it produces. Especially will be tricky for the template shorthand using `*templateDir`

**Comment by wagnermaciel** [comment]
@devversion Ok, I think this should work for all use cases. I basically just gave up on trying to use the ast and instead I'm copying the whitespace directly from the whatever comes after the tag name of the start source span

File: src/material/schematics/ng-generate/mdc-migration/rules/tree-traversal.ts [hunk 1]
```diff
@@ -158,19 +158,24 @@ export function updateAttribute(
...
+  // read and store each char until we reach a non-whitespace char.
 
-  return prefix + attrText + suffix;
+  for (let i = startOffset; i < node.startSourceSpan.end.offset - 1; i++) {
+    if (!/\s/.test(html.charAt(i))) {
+      break;
+    }
+    whitespace += html.charAt(i);
+  }
+  return whitespace || ' ';
```
Comment (devversion) [inline src/material/schematics/ng-generate/mdc-migration/rules/tree-traversal.ts:178]: Should this be two spaces?

**Review by devversion — APPROVED** [review]

File: src/material/schematics/ng-generate/mdc-migration/rules/tree-traversal.ts (hunk 1 above)
Comment (devversion) [reply src/material/schematics/ng-generate/mdc-migration/rules/tree-traversal.ts:178]: 👍

**Pull request merged**